base/pkg/dir.jl

```diff
@@ -37,15 +37,16 @@ function init(meta::AbstractString=DEFAULT_META, branch::AbstractString=META_BRA
     end
     dir = path()
     info("Initializing package repository $dir")
-    temp_dir = mktempdir()
     metadata_dir = joinpath(dir, "METADATA")
     if isdir(metadata_dir)
         info("Package directory $dir is already initialized.")
         Git.set_remote_url(meta, dir=metadata_dir)
         return
     end
+    local temp_dir
     try
-        mkpath(temp_dir)
+        mkpath(dir)
```

Comment:
Won't this line fail if `~/.julia/v0.4` exists but not `~/.julia/v0.4/METADATA`?